_drafts/SC2-1-2-no-keyboard-trap-non-standard-navigation.md

```diff
@@ -0,0 +1,153 @@
+
+### Expectation 1
+
+For each target element help information is [visible on the page][] and [exposed to assistive technologies][] or can be navigated to from within the keyboard trap.
+
+**Note**: As per Success Criterion 2.1.1 Keyboard the help information should be accessible through a keyboard interface. This however is the test subject for another ACT rule.
+
+### Expectation 2
+
+The help information explains how to cycle to the browser UI, or on how to get to a point from where it is possible to cycle to the browser UI, using [standard keyboard navigation][].
+
+### Expectation 3
+
+For each target element focus can cycle to the browser UI by using the method advised in the help information.
+
+**Note**: Cycling back to the browser UI can be done both by moving forward through the tab order and by moving backwards. It is not possible to fulfil this expectation by using browser specific shortcuts to return to the browser UI.
+
+## Assumptions
+
+- The WCAG success criterion applies to all content where focus can be moved to through keyboard navigation.
+- It is not possible to use unmodified arrow or tab keys, or other standard exit methods to move focus away.
+- The focus order in keyboard navigation is cyclical, not linear, meaning that the focus order will cycle to the first/last element when it moves away from the last/first element.
+
+## Accessibility support
+
+There are no major accessibility support issues known for this rule.
+
+## Background
+
+- https://www.w3.org/TR/UNDERSTANDING-WCAG20/keyboard-operation-trapping.html
+- https://www.w3.org/TR/2016/NOTE-WCAG20-TECHS-20161007/G21
+
+## Test Cases
+
+### Passed
+
+```html
+<!-- Keyboard trap with help information in a paragraph before, and where the method adviced works -->
+
+<script>
+var trapOn = false ;
+</script>
+
+<p class="target">Press the M-key to Exit</p>
+<a id="link1" href="#">Link 1</a>
+<button id="btn1" class="target" onblur="(function(e){trapOn=true; document.getElementById('btn2').focus();})(event)">Button 1</button>
+<button id="btn2" class="target" onkeydown="(function(e){ if (e.keyCode === 77){trapOn=false;document.getElementById('link2').focus();}})(event)" onblur="(function(e){ if(trapOn){document.getElementById('btn1').focus();}})(event)">Button 2</button>
+<a id="link2" href="#">Link 2</a>
+```
+
+```html
+<!-- Keyboard trap with help information within the trap, and where the method adviced works --> 
+
+<script>
+var trapOn = false ;
+</script>
+
+<a id="link1" href="#">Link 1</a>
+<button id="btn1" class="target" onblur="(function(e){trapOn=true; document.getElementById('btn2').focus();})(event)">Button 1</button>
+<p class="target">Press the M-key to Exit</p>
+<button id="btn2" class="target" onkeydown="(function(e){ if (e.keyCode === 77){trapOn=false;document.getElementById('link2').focus();}})(event)" onblur="(function(e){ if(trapOn){document.getElementById('btn1').focus();}})(event)">Button 2</button>
+<a id="link2" href="#">Link 2</a>
+````
+
```

Comment:
Add a test case where there is a "confirm" button that can be removed with "enter" key.

Reply:
We don't need this test case for this rule, as Enter is a standard way of keyboard navigation.